Mathlib/MeasureTheory/Integral/DominatedConvergence.lean

```diff
@@ -516,6 +516,44 @@ nonrec theorem _root_.MeasureTheory.Integrable.continuous_primitive (h_int : Int
   continuous_primitive (fun _ _ => h_int.intervalIntegrable) a
 #align measure_theory.integrable.continuous_primitive MeasureTheory.Integrable.continuous_primitive
 
+variable [IsLocallyFiniteMeasure μ] [LocallyCompactSpace X] {f : X → ℝ → E}
```

Comment:
I don't understand why you would need a local compactness assumption for this theorem. Using `IsCompact.mem_uniformity_of_prod`, you should be able to get rid of this assumption (see for instance how it's done in `continuousOn_integral_bilinear_of_locally_integrable_of_compact_support`).

Reply:
I had checked that the proof as written requires it, but not tried to deeply dissect it. I'll try to remove the assumption; thanks for the pointer!

Reply:
I noticed `WeaklyLocallyCompactSpace` is sufficient, but I guess you wanted to drop that as well.

Reply:
I've changed the proof to remove the local compactness (and also first countability). It's a complete rewrite, so it will need another reviewer to check it.